### PR TITLE
Improve user input filtering on user settings page and for task tags [SCI-4707]

### PR DESCRIPTION
**Pull request by okriuchykhin**
Jira ticket: [SCI-4707](https://biosistemika.atlassian.net/browse/SCI-4707)

### What was done
Improve user input filtering on user settings page and for task tags

### CODE REVIEW

File: app/assets/javascripts/shared/inline_editing.js [hunk 1]
```diff
@@ -80,7 +81,7 @@ var inlineEditing = (function() {
         }
 
         container.find('.view-mode')
-          .html(viewData)
+          .text(viewData)
```
Comment (aignatov-bio) [inline app/assets/javascripts/shared/inline_editing.js:84]: For smart annotation fields, we still need to use html. You can check if the field for smart annotation `container.data('smart-annotation')`. As I remember now we use it in comments.

File: app/controllers/my_module_tags_controller.rb [hunk 2]
```diff
@@ -112,18 +114,20 @@ def destroy
...
-    if !(tags.map { |i| i[:label] }.include? params[:query]) && !params[:query].empty?
-      tags = [{ value: 0, label: params[:query], params: { color: nil } }] + tags
+    all_tags = @my_module.experiment.project.tags
+    tags = all_tags.where.not(id: assigned_tags)
+                   .search(current_user, false, params[:query])
+                   .select(:id, :name, :color)
+                   .limit(6)
+
+    tags = tags.map do |tag|
+      { value: tag.id, label: sanitize_input(tag.name), params: { color: sanitize_input(tag.color) } }
```
Comment (aignatov-bio) [inline app/controllers/my_module_tags_controller.rb:124]: Colors predefined list in constants

Comment (okriuchykhin) [reply app/controllers/my_module_tags_controller.rb:124]: And? Please give me a link to the lines where this is enforced?

Comment (aignatov-bio) [reply app/controllers/my_module_tags_controller.rb:124]: Well, yes someone can force set custom color. But what he can do with 7 characters?

Comment (okriuchykhin) [reply app/controllers/my_module_tags_controller.rb:124]: And if later this limit is changed, what will happen to this code?

Comment (aignatov-bio) [reply app/controllers/my_module_tags_controller.rb:124]: In this case yes, it can become issue

**Pull request merged**